### PR TITLE
feat: migrate to python for data validation

### PR DESCRIPTION
**Pull request by onedr0p**
This completely removes Ansible for data validation of the `config.yaml` and instead wraps validation while running `makejinja`.

Important Changes:

- `task init` installs the venv
- Ansible `requirements.yaml` and `requirements.txt` are templated to jinja and moved into the `bootstrap/templates/ansible` folder
- `task ansible:deps` installs the specific Ansible deps for the venv

Todo:

- [x] Add validation for `bootstrap_nodes`
- [ ] ~~Add validation for checking if the age key exists on filesystem~~ (done is taskfile)
- [x] Update readme

### CODE REVIEW

File: bootstrap/scripts/validation.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+GLOBAL_CLI_TOOLS          = ["age", "cloudflared", "flux", "sops", "jq", "kubeconform", "kustomize"]
+TALOS_CLI_TOOLS           = ["talosctl", "talhelper"]
+K0S_CLI_TOOLS             = ["k0sctl"]
+
+def required(*keys: str):
+    def wrapper_outter(func: Callable):
+        @wraps(func)
+        def wrapper(data: dict, *args, **kwargs) -> None:
+            for key in keys:
+                if data.get(key) is None:
```
Comment (brunnels) [inline bootstrap/scripts/validation.py:24]: `if not data.get(key):` is better

File: bootstrap/scripts/validation.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+def validate_python_version() -> None:
+    required_version = (3, 11, 0)
+    if sys.version_info < required_version:
+        raise ValueError(f"Python version is below 3.11. Please upgrade.")
+
+@required("bootstrap_distribution")
+def validate_cli_tools(distribution: str) -> None:
+    distro = _validate_distribution(distribution)
+    for tool in GLOBAL_CLI_TOOLS:
+        if which(tool) is None:
```
Comment (brunnels) [inline bootstrap/scripts/validation.py:58]: `if not which(tool):`

File: bootstrap/scripts/validation.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+@required("bootstrap_acme_email", "bootstrap_acme_production_enabled")
+def validate_acme_email(email: str, acme_production: bool) -> None:
+    try:
+        validate_email(email)
+    except EmailNotValidError as e:
+        raise ValueError(f"Invalid ACME email {email}") from e
+
+@required("bootstrap_flux_github_webhook_token")
+def validate_flux_github_webhook_token(token: str) -> None:
+    if re.match(r"^[a-zA-Z0-9]+$", token) is None:
```
Comment (brunnels) [inline bootstrap/scripts/validation.py:128]: `if not re.match(...):`

File: bootstrap/scripts/validation.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        raise ValueError(f"Invalid Flux GitHub webhook token {token}")
+
+@required("bootstrap_cloudflare_domain", "bootstrap_cloudflare_token", "bootstrap_cloudflare_account_tag", "bootstrap_cloudflare_tunnel_secret", "bootstrap_cloudflare_tunnel_id")
+def validate_cloudflare(domain: str, token: str, account_tag: str, tunnel_secret: str, tunnel_id: str) -> None:
+    try:
+      cf = CloudFlare.CloudFlare(token=token)
+      zones = cf.zones.get(params={"name": domain})
+      if len(zones) == 0:
```
Comment (brunnels) [inline bootstrap/scripts/validation.py:136]: `if not zones:`

File: bootstrap/scripts/validation.py (hunk 1 above)
Comment (onedr0p) [reply bootstrap/scripts/validation.py:24]: I get an error with that change:

```
  File "/Users/devin/Code/repos/onedr0p/flux-template-cluster/bootstrap/scripts/loader.py", line 29, in __init__
    validation.validate_acme_email(data)
  File "/Users/devin/Code/repos/onedr0p/flux-template-cluster/bootstrap/scripts/validation.py", line 25, in wrapper
    raise ValueError(f"Missing required key {key}")
ValueError: Missing required key bootstrap_acme_production_enabled
task: Failed to run task "configure": exit status 1
```

Comment (brunnels) [reply bootstrap/scripts/validation.py:24]: Okay, if it breaks then go with what you have.  👍

**Pull request merged**